cyclonedx/output/json.py

```diff
@@ -56,28 +56,44 @@ def generate(self, force_regeneration: bool = False) -> None:
         if self.generated and not force_regeneration:
             return
 
+        self.get_bom().validate()
+
         schema_uri: Optional[str] = self._get_schema_uri()
         if not schema_uri:
             raise FormatNotSupportedException(
-                f'JSON is not supported by CycloneDX in schema version {self.schema_version.to_version()}'
-            )
-
-        vulnerabilities: Dict[str, List[Dict[Any, Any]]] = {"vulnerabilities": []}
-        if self.get_bom().components:
-            for component in cast(List[Component], self.get_bom().components):
-                for vulnerability in component.get_vulnerabilities():
-                    vulnerabilities['vulnerabilities'].append(
-                        json.loads(json.dumps(vulnerability, cls=CycloneDxJSONEncoder))
-                    )
+                f'JSON is not supported by CycloneDX in schema version {self.schema_version.to_version()}')
+
+        extras = {}
+        if self.bom_supports_dependencies():
+            dependencies: List[Dict[str, Union[str, List[str]]]] = []
+            if self.get_bom().metadata.component:
+                dependencies.append({
+                    'ref': str(cast(Component, self.get_bom().metadata.component).bom_ref),
```

Comment:
could you expain wjy `cast` is needed ?  did not try it myself, so i do not have any idea.
if it is just for the typing, it could help to store `self.get_bom()` result in a variable, so `mypy` sees that the variables are constant and type-ckeched already via `if bom.metadata.component:`

Reply:
@jkowalleck `mypy` was complaining about `self.get_bom().metadata.component` as this is defined as `Optional[Component]`.

I would have thought given this code is inside an `if self.get_bom().metadata.component`, `mypy` would have been happy, but it was not without the `cast()`.

Thoughts?